lib/utils/resource-links.js

```diff
@@ -0,0 +1,60 @@
+function isResourceLink (field) {
+  return field && field.sys && field.sys.type === 'ResourceLink'
+}
+
+function getResourceLinkFields (data) {
+  const items = [].concat(...(data.items || []), ...Object.values(data.includes || {}).flat())
+  const fields = items.flatMap(({ fields }) => Object.values(fields).flat())
+  return fields.filter(isResourceLink)
+}
+
+function parseResourceLink (resourceLink) {
+  const urn = resourceLink.sys.urn
+  // For now just assume every resourceLink has type Contentful:Entry
+  const [, spaceId, , entryId] = urn.split('/')
+  return { spaceId, entryId, urn }
+}
+
+function groupBy (key, objArray) {
+  return objArray.reduce((acc, obj) => {
+    const objKey = obj[key]
+    acc[objKey] = (acc[objKey] || []).concat(obj)
+    return acc
+  }, {})
+}
+
+async function getEntitiesFromSpace (spaceId, entries, extraApis, include) {
+  const apiInstance = extraApis[spaceId]
+  if (!apiInstance) {
+    return []
+  }
+
+  const entryIds = entries.map(({ entryId }) => entryId)
+
+  let response = { items: [] }
+  const uniqueEntryIds = [...new Set(entryIds)]
+  response = await apiInstance.getEntries({ 'sys.id[in]': uniqueEntryIds.join(','), include }).catch(() => ({}))
+  return response.items || []
```

Comment:
We should probably not only return the `items` but also the `includes` from the response (for assets and same-space Links). This would probably mean changing the return type which will require some refactoring in `addResourceLinksToIncludes`
```suggestion
  return {...(response.includes || {}), Entry: {...((response.includes && response.includes.Entry) || []), ...(response.items || [])}
```

Reply:
I was incorrect, I forgot that the `apiInstance` will call it's own `wrapEntryCollection` function to copy the includes into the field key